test/github-issues/528/issue-528.ts

```diff
@@ -0,0 +1,57 @@
+import "reflect-metadata";
+import {createTestingConnections, closeTestingConnections, reloadTestingDatabases} from "../../utils/test-utils";
+import {Connection} from "../../../src/connection/Connection";
+
+import {Post} from "./entity/Post";
+import {Author} from "./entity/Author";
+
+describe("github issues > #528 Migrations failing on timestamp validation", () => {
+
+    let connections: Connection[];
+    before(async () => connections = await createTestingConnections({
+        entities: [__dirname + "/entity/*{.js,.ts}"],
+        schemaCreate: true,
+        dropSchemaOnConnection: true,
+    }));
+    beforeEach(() => reloadTestingDatabases(connections));
+    after(() => closeTestingConnections(connections));
+
+    it("should success migrating (without failing on timestamp navigation)", () => Promise.all(connections.map(async connection => {
+
+        // first insert all the data
+        const author = new Author();
+        author.firstName = "Artur";
+        author.lastName = "Lavrischev";
+
+        const postOld = new Post();
+        postOld.title = "Hello World!";
+        postOld.author = author;
+
+        const postNew = new Post();
+        postNew.title = "Hello!";
+        postNew.author = author;
+
+        const postRepository = connection.getRepository(Post);
+
```

Comment:
i didn't understand point of your tests... you create two posts, save first one, then second... but you dont create migrations that should run, right? Don't understand what exactly you are testing here...